src/main/java/org/apache/commons/lang3/BooleanUtils.java

```diff
@@ -0,0 +1,1142 @@
+
+    /**
+     * <p>Converts a boolean to a String returning {@code 'yes'}
+     * or {@code 'no'}.</p>
+     *
+     * <pre>
+     *   BooleanUtils.toStringYesNo(true)   = "yes"
+     *   BooleanUtils.toStringYesNo(false)  = "no"
+     * </pre>
+     *
+     * @param bool  the Boolean to check
+     * @return {@code 'yes'}, {@code 'no'}, or {@code null}
+     */
+    public static String toStringYesNo(final boolean bool) {
+        return toString(bool, YES, NO);
+    }
+
+    /**
+     * <p>Converts a Boolean to a String returning {@code 'yes'},
+     * {@code 'no'}, or {@code null}.</p>
+     *
+     * <pre>
+     *   BooleanUtils.toStringYesNo(Boolean.TRUE)  = "yes"
+     *   BooleanUtils.toStringYesNo(Boolean.FALSE) = "no"
+     *   BooleanUtils.toStringYesNo(null)          = null;
+     * </pre>
+     *
+     * @param bool  the Boolean to check
+     * @return {@code 'yes'}, {@code 'no'}, or {@code null}
+     */
+    public static String toStringYesNo(final Boolean bool) {
+        return toString(bool, YES, NO, null);
+    }
+
+    /**
+     * <p>Performs an xor on a set of booleans.</p>
+     *
+     * <pre>
+     *   BooleanUtils.xor(true, true)   = false
+     *   BooleanUtils.xor(false, false) = false
+     *   BooleanUtils.xor(true, false)  = true
+     * </pre>
+     *
+     * @param array  an array of {@code boolean}s
+     * @return the result of the xor operations
+     * @throws NullPointerException if {@code array} is {@code null}
+     * @throws IllegalArgumentException if {@code array} is empty.
+     */
+    public static boolean xor(final boolean... array) {
+        ObjectUtils.requireNonEmpty(array, "array");
+        // false if the neutral element of the xor operator
+        boolean result = false;
+        for (final boolean element : array) {
+            result ^= element;
+        }
+
+        return result;
+    }
+
+    /**
+     * <p>Performs an xor on an array of Booleans.</p>
+     *
+     * <pre>
+     *   BooleanUtils.xor(new Boolean[] { Boolean.TRUE, Boolean.TRUE })   = Boolean.FALSE
+     *   BooleanUtils.xor(new Boolean[] { Boolean.FALSE, Boolean.FALSE }) = Boolean.FALSE
+     *   BooleanUtils.xor(new Boolean[] { Boolean.TRUE, Boolean.FALSE })  = Boolean.TRUE
+     *   BooleanUtils.xor(Boolean.TRUE, Boolean.FALSE, Boolean.FALSE)     = Boolean.TRUE
+     * </pre>
+     *
+     * @param array  an array of {@code Boolean}s
+     * @return the result of the xor operations
+     * @throws NullPointerException if {@code array} is {@code null}
+     * @throws IllegalArgumentException if {@code array} is empty.
+     * @throws IllegalArgumentException if {@code array} contains a {@code null}
+     */
+    public static Boolean xor(final Boolean... array) {
+        ObjectUtils.requireNonEmpty(array, "array");
+        try {
+            final boolean[] primitive = ArrayUtils.toPrimitive(array);
+            return xor(primitive) ? Boolean.TRUE : Boolean.FALSE;
+        } catch (final NullPointerException ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)